Added `quantlib_tools` package, test case and command

def-mycroft opened the pull request:
Added a package that provides some helper funcs for `QuantLib`. There is a test case in for the options model in the `quantlib_tools` package and a command in the base controller allows for modeling the price of a single option.

def-mycroft committed:
Added quantlib tools package, requirements

Added quantlib_tools package and modified requirements for support.

Full testing is not yet configured, but was able to successfully pass
test cases present in twcli/core/quantlib_tools/test_models.py.

An error was raised when trying to build the environment for testing,
the package pytest and a couple of others were listed in both
requirements.txt and requirements-dev.txt. The duplicated packages were
removed, which suppressed the error.

def-mycroft committed:
Added command for option price model

Added a simple cli tool for modeling options price. Made a minor change
in the date format in the test.

Graeme22 commented:
Looks good, thanks for adding the test. Let me get a CI pipeline running, then I'll merge this.

def-mycroft commented:
cool. I haven't done anything with the cement testing apparatus yet, just
that unit test module in core.quantlib_tools.

we might want to talk about the project layout, when I've used cement I
organize everything into packages under the "core" folder, seems to work
pretty well, but you might have a better idea.

On Thu, May 13, 2021 at 3:12 PM Graeme Holliday ***@***.***>
wrote:

> Looks good, thanks for adding the test. Let me get a CI pipeline running,
> then I'll merge this.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Graeme22/tastyworks-cli/pull/7#issuecomment-840805241>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEY7CLBUVXBMIBVRTJXSEHTTNQXDDANCNFSM443EEJNQ>
> .
>


-- 
Joseph Dasenbrock

Graeme22 committed:
Create python-publish.yml

Graeme22 committed:
Merge pull request #10 from Graeme22/Graeme22-patch-2

Create python-publish.yml

Graeme22 committed:
Linting/testing template (#9)

* Linting/testing template

* CI/CD

* more CI/CD setup

* formatting base code

* flake8 fix

* flake8 flags

* workflow change

* install requirements-dev as well

* remove flags

* modify workflow

* bad formatting example

* fix errors, update action

* change workflow

* remove mypy (no types for cement)

* add path to isort

* isort fix

* flake8 is working

* should be good to go, we just need working tests

* hopefully we can pass tests now

* pytest fix

Co-authored-by: Graeme Holliday <graeme@garch.localdomain>

Graeme22 commented:
Hi @def-mycroft, please rebase and add a commit to this, that will trigger the CI test.

Graeme22 commented:
Also, please move the code into the `twcli/core/quant` folder.

Graeme22 committed:
structure for future work (#11)

Co-authored-by: Graeme Holliday <graeme@garch.localdomain>

Graeme22 committed:
cement module creation, formatting (#12)

* structure for future work

* cement formatting & project structure

Co-authored-by: Graeme Holliday <graeme@garch.localdomain>

def-mycroft committed:
Merge branch 'main' of https://github.com/def-mycroft/tastyworks-cli into def-mycroft-main

Fixed requirements

def-mycroft committed:
Merge branch 'def-mycroft-main' into main

def-mycroft committed:
Fixed duplicated pandas line in requirements

def-mycroft committed:
Modified requirements, refactored quantlib_tools

Modified requirements files to exclude several duplicated values between
requirements.txt and requirements-dev.txt, these were probably created
after merging.

Refactored modules in quantlib_tools in order for linting tests to pass.

def-mycroft commented:
hi @Graeme22 , the tests passed, I'm not sure what to do now, do you have to merge the pr? thanks

def-mycroft commented:
also @Graeme22 , previously you said "Also, please move the code into the `twcli/core/quant` folder", should I move the `quantlib_tools` folder there or the files in `quantlib_tools`? I tried moving the `quantlib_tools` folder there, but then the tests didn't run (for that code).

Graeme22 commented:
> also @Graeme22 , previously you said "Also, please move the code into the `twcli/core/quant` folder", should I move the `quantlib_tools` folder there or the files in `quantlib_tools`? I tried moving the `quantlib_tools` folder there, but then the tests didn't run (for that code).

I think moving the scripts to the `quant` folder seems the cleanest.

Notice how at the top of this PR it says "Files changed:  48"? That's no bueno. You need to rebase this with the current master.
https://anavarre.net/how-to-rebase-a-github-pull-request/#:~:text=Rebase%20your%20pull%20request&text=First%20things%20first%2C%20we%20need,we%20originally%20created%20the%20PR.&text=Now%2C%20we%20can%20run%20the,top%20of%20our%20development%20branch.&text=The%20last%20remaining%20step%20is,does%20replace%20the%20outdated%20PR.

def-mycroft commented:
> > also @Graeme22 , previously you said "Also, please move the code into the `twcli/core/quant` folder", should I move the `quantlib_tools` folder there or the files in `quantlib_tools`? I tried moving the `quantlib_tools` folder there, but then the tests didn't run (for that code).
> 
> I think moving the scripts to the `quant` folder seems the cleanest.
> 
> Notice how at the top of this PR it says "Files changed: 48"? That's no bueno. You need to rebase this with the current master.
> https://anavarre.net/how-to-rebase-a-github-pull-request/#:~:text=Rebase%20your%20pull%20request&text=First%20things%20first%2C%20we%20need,we%20originally%20created%20the%20PR.&text=Now%2C%20we%20can%20run%20the,top%20of%20our%20development%20branch.&text=The%20last%20remaining%20step%20is,does%20replace%20the%20outdated%20PR.

I don't see how to rebase with the current master, I get an error `invalid upstream`... [here](https://paste.ubuntu.com/p/y4ppDbZk2H/) is a pastebin. 

I do already have the latest commit from the main repo in my fork... (30f92c63)

Graeme22 commented:
> I don't see how to rebase with the current master, I get an error `invalid upstream`... [here](https://paste.ubuntu.com/p/y4ppDbZk2H/) is a pastebin.
> 
> I do already have the latest commit from the main repo in my fork... ([30f92c6](https://github.com/Graeme22/tastyworks-cli/commit/30f92c6378d536cc0a9a74382bc10262b829abbe))

One thing that might make it easier for you is to just open a new pull request on a branch from the current `main` and add your changes there.

def-mycroft commented:
okay, I'll just close this one then.
